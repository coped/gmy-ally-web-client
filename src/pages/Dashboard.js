import React from "react";

export default function Dashboard() {
  return <p>This is the dashboard.</p>;
}
