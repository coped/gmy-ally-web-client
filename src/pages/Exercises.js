import React, { useEffect, useState } from "react";
import { Api } from "lib/api";
import { PageLoader } from "components/common";
import { useExercises } from "context/exercises";
import "bulma-quickview/dist/css/bulma-quickview.min.css";
import bulmaQuickview from "bulma-quickview/dist/js/bulma-quickview.min.js";

export default function Exercises() {
  const {
    allExercises,
    muscleGroups,
    categorizedExercises,
    setAllExercises,
  } = useExercises();
  const [loading, setLoading] = useState(false);
  const [currentMuscleGroup, setCurrentMuscleGroup] = useState("abdominals");
  const [quickviewExercise, setQuickviewExercise] = useState();

  // Fetch exercise data on component mount
  useEffect(() => {
    async function fetchData() {
      setLoading(true);
      const data = await Api.indexExercises();
      setAllExercises(data.payload.exercises);
      setQuickviewExercise(data.payload.exercises[0]);
      setLoading(false);
    }
    !allExercises && fetchData();
  }, [allExercises, setAllExercises]);

  useEffect(() => {
    if (allExercises && !quickviewExercise)
      setQuickviewExercise(allExercises[0]);
    if (quickviewExercise) bulmaQuickview.attach();
  });

  function showExercises() {
    console.log(allExercises);
    allExercises && console.log(categorizedExercises);
  }

  function capitalize(string) {
    return string[0].toUpperCase() + string.substring(1);
  }

  return (
    <div id="Exercises">
      <PageLoader loading={loading} className="is-info" />
      {quickviewExercise && (
        <div id="quickviewDefault" className="quickview">
          <header className="quickview-header">
            <p className="title">{quickviewExercise.title}</p>
            <span className="delete" data-dismiss="quickview"></span>
          </header>

          <div className="quickview-body">
            <div className="quickview-block section">
              <h3 className="is-size-5">Movement type:</h3>
              <p>{quickviewExercise.movement_type}</p>
              <h3 className="is-size-5">Equipment:</h3>
              <div className="content">
                <ul>
                  {quickviewExercise.equipment.split("\n").map((item) => (
                    <li key={item}>{capitalize(item)}</li>
                  ))}
                </ul>
              </div>
              <h3 className="is-size-5">Steps:</h3>
              <div className="content">
                <ol>
                  {quickviewExercise.steps.split("\n").map((step, index) => (
                    <li key={"step-" + index}>{step}</li>
                  ))}
                </ol>
              </div>
            </div>
          </div>

          <footer className="quickview-footer">GYM PARTNER</footer>
        </div>
      )}
      <div className="columns">
        <div className="column is-3">
          <aside className="menu">
            <p className="menu-label">SEARCH EXERCISES BY MUSCLE GROUP</p>
            <ul className="menu-list">
              {muscleGroups &&
                muscleGroups.map((name) => (
                  <li key={name}>
                    <a onClick={() => setCurrentMuscleGroup(name)}>
                      {capitalize(name)}
                    </a>
                  </li>
                ))}
            </ul>
          </aside>
        </div>
        <div className="column">
          <h3 className="is-size-3 has-text-centered">
            {capitalize(currentMuscleGroup)}
          </h3>
          <div className="box">
            <ul className="menu-list">
              {categorizedExercises &&
                categorizedExercises[currentMuscleGroup].map((exercise) => (
                  <li className="list-item" key={exercise.name}>
                    <a
                      onClick={() => setQuickviewExercise(exercise)}
                      data-show="quickview"
                      data-target="quickviewDefault"
                    >
                      {exercise.title}
                    </a>
                  </li>
                ))}
            </ul>
          </div>
        </div>
      </div>
      <button onClick={showExercises} className="button">
        Show exercises
      </button>
    </div>
  );
}
