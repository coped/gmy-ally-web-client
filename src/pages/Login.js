import React, { useState } from "react";
import { Link, Redirect } from "react-router-dom";
import { FormInput, FormButton } from "components/form";
import { Notification } from "components/common";
import { useAuth } from "context/auth";
import { useUser } from "context/user";
import Api from "lib/api";
import "assets/Login.scss";

export default function Login() {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [apiMessages, setApiMessages] = useState([]);
  const [form, setForm] = useState({ email: "", password: "" });

  const { setAuthContext } = useAuth();
  const { setUserContext } = useUser();

  function onChange(event) {
    const target = event.target;
    setForm({
      ...form,
      [target.name]: target.value,
    });
  }

  async function loginUser(e) {
    e.preventDefault();
    setIsLoading(true);
    const data = await Api.login({ info: form });
    if (data.status === "success") {
      setAuthContext(data.payload.jwt);
      setUserContext(data.payload.user);
      setIsLoggedIn(true);
    } else {
      setApiMessages(data.messages);
      setIsLoading(false);
    }
  }

  if (isLoggedIn) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <div id="Login">
      <h1 className="login-title">Log in</h1>
      <form onSubmit={loginUser} className="column is-6 box">
        {apiMessages &&
          apiMessages.map((response, index) => (
            <Notification key={index} classList={[`is-${response.type}`]}>
              <p>{response.message}</p>
            </Notification>
          ))}
        <FormInput
          id="email-input"
          label="Email:"
          name="email"
          type="email"
          placeholder="your@email.com"
          value={form.email}
          onChange={onChange}
        />
        <FormInput
          id="password-input"
          label="Password:"
          name="password"
          type="password"
          value={form.password}
          onChange={onChange}
        />
        <FormButton classList={["is-link"]} loading={isLoading}>
          Log in
        </FormButton>
        <p>
          Don't have an account? <Link to="/signup">Sign up</Link>.
        </p>
      </form>
    </div>
  );
}
