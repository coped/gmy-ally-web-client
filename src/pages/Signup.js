import React, { useState } from "react";
import { Notification } from "components/common";
import { FormInput, FormButton } from "components/form";
import { AsyncRequest, endpoints, Messages } from "lib";
import { Link, Redirect } from "react-router-dom";
import "assets/Signup.scss";
import { useAuth } from "context/auth";

export default function Signup(props) {
  const [isLoading, setIsLoading] = useState(false);
  const [form, setForm] = useState({
    name: "",
    email: "",
    password: "",
    passwordConfirmation: "",
  });
  const [apiMessages, setApiMessages] = useState("");
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const { setAuthToken } = useAuth();

  function signupUser(credentials) {
    setIsLoading(true);
    AsyncRequest.post(endpoints.users.create, credentials)
      .then((response) => response.json())
      .then((data) => {
        if (data.status === "success") {
          setAuthToken(data.payload.jwt)
          setIsLoggedIn(true);
        } else {
          setApiMessages(data.messages);
          setIsLoading(false);
        }
      })
      .catch((error) => {
        setApiMessages([Messages.connectionError]);
        setIsLoading(false);
      });
  }

  function onChange(event) {
    const target = event.target;
    setForm({ ...form, [target.name]: target.value });
  }

  function onFormSubmit(event) {
    event.preventDefault();
    signupUser({
      user: {
        name: form.name,
        email: form.email,
        password: form.password,
        password_confirmation: form.passwordConfirmation,
      },
    });
  }

  if (isLoggedIn) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <div id="Signup">
      <h1 className="signup-title">Sign up</h1>
      <form onSubmit={onFormSubmit} className="column is-6 box">
        {apiMessages &&
          apiMessages.map((message, index) => (
            <Notification key={index} type={message.type}>
              <p>{message.message}</p>
            </Notification>
          ))}
        <FormInput
          id="name-input"
          label="Name:"
          name="name"
          type="text"
          placeholder="Your name"
          value={form.name}
          onChange={onChange}
        />
        <FormInput
          id="email-input"
          label="Email:"
          name="email"
          type="email"
          placeholder="your@email.com"
          value={form.email}
          onChange={onChange}
        />
        <FormInput
          id="password-input"
          label="Password:"
          name="password"
          type="password"
          value={form.password}
          onChange={onChange}
        />
        <FormInput
          id="password-confirmation-input"
          label="Password confirmation:"
          name="passwordConfirmation"
          type="password"
          value={form.passwordConfirmation}
          onChange={onChange}
        />
        <FormButton classList={["is-link"]} loading={isLoading}>
          Sign up
        </FormButton>
        <p className="is-size-6">
          Already have an account? <Link to="/login">Log in</Link>.
        </p>
      </form>
    </div>
  );
}
