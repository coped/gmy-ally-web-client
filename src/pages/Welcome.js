import React from "react";
import "assets/Welcome.scss";
import { Button } from "components/common";
import { Link } from "react-router-dom";

export default function Welcome() {
  return (
    <div id="Welcome">
      <Link to="/signup">
        <Button classList={["is-primary"]}>Sign up</Button>
      </Link>
      <Link to="/login">
        <Button classList={["is-link"]}>Log in</Button>
      </Link>
    </div>
  );
}
