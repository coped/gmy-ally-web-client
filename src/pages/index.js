import Login from "pages/Login";
import Signup from "pages/Signup";
import Welcome from "pages/Welcome";
import Dashboard from "pages/Dashboard";
import Exercises from "pages/Exercises";

export { Login, Signup, Welcome, Dashboard, Exercises };
