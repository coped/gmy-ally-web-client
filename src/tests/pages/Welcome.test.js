import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Welcome } from "pages";
import { MemoryRouter } from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe("Welcome", () => {
  const withContext = (component) => (
    <MemoryRouter initialEntries={["/"]}>{component}</MemoryRouter>
  );

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(withContext(<Welcome />), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(withContext(<Welcome />));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
