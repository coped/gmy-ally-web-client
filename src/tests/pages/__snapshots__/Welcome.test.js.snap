// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome has a valid snapshot 1`] = `
<div
  id="Welcome"
>
  <a
    href="/signup"
    onClick={[Function]}
  >
    <button
      className="button is-primary"
    >
      Sign up
    </button>
  </a>
  <a
    href="/login"
    onClick={[Function]}
  >
    <button
      className="button is-link"
    >
      Log in
    </button>
  </a>
</div>
`;
